megamek/src/megamek/client/AbstractClient.java

```diff
@@ -359,7 +361,7 @@ protected synchronized void checkDuplicateNamesDuringAdd(Entity entity) {
         }
     }
 
-    protected void receiveUnitReplace(Packet packet) {
+    protected void receiveUnitReplace(Packet packet) throws InvalidPacketDataException {
         @SuppressWarnings(value = "unchecked") List<Force> forces = (List<Force>) packet.getObject(1);
         forces.forEach(force -> getGame().getForces().replace(force.getId(), force));
 
```

Comment:
There should be a getForceList method on packet to remove the unchecked suppression.